core/tests/webdriverio_utils/ContributorDashboardAdminPage.js

```diff
@@ -232,7 +236,147 @@ var ContributorDashboardAdminPage = function () {
       `.e2e-test-language-selector-option=${language}`
     );
     await action.click(`${language} option selector`, selectorOption);
-    await action.click('Language Selector', languageSelector);
+  };
+
+  this.navigateThroughMonths = async function (numberOfMonths) {
+    /**
+    We require to use element selector classes without "e2e-*" fix
+    because the mat-datepicker that we use here do not contain any
+    such classes for e2e test purposes. Also we can not add any e2e
+    test class to any ansector element since, the ancestor to this
+    is the body tag, in the DOM.
+    **/
+    var nextMonthButton = $('.mat-calendar-next-button');
+    var prevMonthButton = $('.mat-calendar-previous-button');
+    var calenderBody = $('.mat-calendar-content')
+      .$('.mat-calendar-table')
+      .$('.mat-calendar-body')
+      .$('tr')
+      .$('td');
+
+    var months = [
+      'JAN',
+      'FEB',
+      'MAR',
+      'APR',
+      'MAY',
+      'JUN',
+      'JUL',
+      'AUG',
+      'SEP',
+      'OCT',
+      'NOV',
+      'DEC',
+    ];
+
+    var monthText = await calenderBody.getText();
+    var monthIndex;
+    for (let i = 0; i < 12; i++) {
+      if (months[i] === monthText) {
+        monthIndex = i;
+      }
+    }
+
+    if (numberOfMonths > 0) {
+      for (let i = 0; i < numberOfMonths; i++) {
+        await action.click('Next Month Button', nextMonthButton);
+        if (monthIndex === 11) {
+          monthIndex = 0;
+        } else {
+          monthIndex++;
+        }
+        monthText = await calenderBody.getText();
+        expect(months[monthIndex]).toBe(monthText);
```

Comment:
This isn't quite what I meant in my earlier comment. We want the test to wait for the next month to appear, but here the test will fail if the next month does't show up immediately.

Can you instead wait for the next month to appear? This will allow time for the next month to load in case it's slow, but if the next month never appears, the test will still fail.

ditto below

Reply:
hello @U8NWXD, for writing such a test we need to figure out an element to wait for, whenever next/previous month button is clicked. This element must contain an attribute or class that is unique to a month.
I am wondering is it possible to use `ng-reflect-label` for this purpose:
![Screenshot (153)](https://github.com/oppia/oppia/assets/92575005/aac3922c-9bf7-4adb-9ebf-eb9dadf4f5ce)
Earlier I tried to select this element `ng-reflect-label` and it worked fine on my local machine but failed during GitHub CI check.

Reply:
After some research I found a way to select container using month code.